casepro/contacts/models.py

```diff
@@ -316,13 +318,12 @@ def get_or_create_from_urn(cls, org, urn, name=None):
         Gets an existing contact or creates a new contact. Used when opening a case without an initial message
         """
         normalized_urn = URN.normalize(urn)
-
         contact = cls.objects.filter(urns__contains=[normalized_urn]).first()
         if not contact:
             URN.validate(normalized_urn)
-
+            uuid = org.get_backend().push_contact(org, normalized_urn)
             contact = cls.objects.create(org=org, name=name, urns=[normalized_urn], is_stub=False)
-            org.get_backend().push_contact(org, contact)
+            contact.uuid = uuid
```

Comment:
I think you need to save the contact here, otherwise this change doesn't get written to the database.
but can you add the uuid to the `create` call above rather? does that work?

Reply:
@megh-k I think it would be better to add the `uuid` to the `create` call. that way if there's a unique restriction on `uuid` then it get's picked up before you create the object